### DISCUSSION TITLE
asynchronous programming

### DISCUSSION THREAD

**Discussion by Kirillbiliashov** [HowProgrammingWorks/LiveQA#15] [Q&A]
Почему считается, что вызов функции с ключевым словом await не блокирует поток событий, хотя любой код написанный после этого вызова будет ждать выполнения функции с ключевым словом await? (например в коде 
```js
(async () => {
  await sleep(3000);
  console.log('some text');
})();
```
в командную строку ничего не выведется в первые 3 секунды, что можно считать блокировкой потока событий, насколько я понял)

**Comment by tshemsedinov** [comment]
Потому, что во время исполнения `await` может работать другой код, т.е. `await` не блокирует выполнение событий, асинхронных операций, происходящих по всему приложению.

**Reply by vpoliatychko** [reply]
Ответ и рассмотрение этого куска кода есть в этой лекции (таймкод присутсвует): https://youtu.be/Jdf_tZuJbHI?t=805
